app/src/main/java/com/cortado/mafkir/ui/adapter/MafkirContactListAdapter.kt

```diff
@@ -0,0 +1,63 @@
+package com.cortado.mafkir.ui.adapter
+
+import android.graphics.Color
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.navigation.findNavController
+import androidx.navigation.fragment.FragmentNavigatorExtras
+import androidx.recyclerview.widget.RecyclerView
+import com.cortado.mafkir.databinding.MafkircontactItemsBinding
+import com.cortado.mafkir.model.TimeConverter
+import com.cortado.mafkir.persistence.MafkirContact
+import com.cortado.mafkir.ui.ListFragmentDirections
+import com.cortado.mafkir.ui.extension.toTransitionGroup
+import kotlinx.android.synthetic.main.mafkircontact_items.view.*
+
+class MafkirContactListAdapter() : RecyclerView.Adapter<MafkirContactListAdapter.ViewHolder>() {
+
+    private val mafkirContacts = mutableListOf<MafkirContact>()
```

Comment:
if i understand correctly, on every change of the list, `submitList` should be called and whole recycler will be re-rendered. recyclers are very heavy and can easily become inefficient and make the app stuck, think about modifying only what's necessary. finding the diff through code is a nice option, use `notifyItemChanged` instead of the inefficient `notifyDataSetChanged` when possible

Reply:
d